api/tests/unit/app_analytics/test_tasks.py

```diff
@@ -0,0 +1,318 @@
+@pytest.mark.django_db(databases=["analytics"])
+def test_populate_api_usage_bucket_15m_bucket(freezer):
+    # Given
+    environment_id = 1
+    bucket_size = 15
+    now = timezone.now()
+    # let's create events at every 1 minutes
+    # for the last two hours, i.e: from 9:09:47 to 7:10:47
+    for i in range(60 * 2):
+        _create_api_usage_event(environment_id, now - timezone.timedelta(minutes=1 * i))
+        # create events in some other environments as well - just to make sure
+        # we don't aggregate them in the same environment
+        _create_api_usage_event(999, now - timezone.timedelta(minutes=1 * i))
+
+    # Next, let's go 1 hr back in the past and run this
+    freezer.move_to(timezone.now() - timezone.timedelta(hours=1))
+    populate_api_usage_bucket(bucket_size, run_every=timedelta(minutes=60))
+
+    # Then - we should have four buckets
+    buckets = (
+        APIUsageBucket.objects.filter(bucket_size=15, environment_id=environment_id)
+        .order_by("created_at")
+        .all()
+    )
+
+    assert len(buckets) == 4
+    assert buckets[0].created_at.isoformat() == "2023-01-19T07:00:00+00:00"
+    # since we started at 7:10:47, the first bucket should have 5 events
+    assert buckets[0].total_count == 5
+
+    # the rest of them should have 15 events
+    assert buckets[1].created_at.isoformat() == "2023-01-19T07:15:00+00:00"
+    assert buckets[1].total_count == 15
+
+    assert buckets[2].created_at.isoformat() == "2023-01-19T07:30:00+00:00"
+    assert buckets[2].total_count == 15
+
+    assert buckets[3].created_at.isoformat() == "2023-01-19T07:45:00+00:00"
+    assert buckets[3].total_count == 15
+
+    # Now, let's move forward 1hr and run this again
+    freezer.move_to(timezone.now() + timezone.timedelta(hours=1))
+    populate_api_usage_bucket(bucket_size, run_every=timedelta(minutes=60))
+
+    # Then - we should have another four buckets created by the second run
+    buckets = (
+        APIUsageBucket.objects.filter(bucket_size=15, environment_id=environment_id)
+        .order_by("created_at")
+        .all()
+    )
+    assert len(buckets) == 8
+
+    # with 15 events each
+    assert buckets[4].created_at.isoformat() == "2023-01-19T08:00:00+00:00"
+    assert buckets[4].total_count == 15
+
+    assert buckets[5].created_at.isoformat() == "2023-01-19T08:15:00+00:00"
+    assert buckets[5].total_count == 15
+
+    assert buckets[6].created_at.isoformat() == "2023-01-19T08:30:00+00:00"
+    assert buckets[6].total_count == 15
+
+    assert buckets[7].created_at.isoformat() == "2023-01-19T08:45:00+00:00"
+    assert buckets[7].total_count == 15
+
+
+@pytest.mark.parametrize(
+    "bucket_size, runs_every_minutes",
+    [(15, 60), (10, 60), (10, 30), (30, 30), (60, 60), (10, 10), (60, 60 * 4)],
+)
+@pytest.mark.freeze_time("2023-01-19T09:09:47.325132+00:00")
+@pytest.mark.django_db(databases=["analytics"])
+def test_populate_api_usage_bucket(freezer, bucket_size, runs_every_minutes):
```

Comment:
I'm unsure why we have these 2 separate tests (although I'm not necessarily complaining) but one is called `test_populate_api_usage_bucket_15m_bucket` which seems to test more specifics about the buckets and this one is just called `test_populate_api_usage_bucket` which just tests that the correct number of buckets are created with the correct start times. Should we add some more documentation here about what each test's purpose is?

Reply:
Yeah, I have updated the name